packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -382,16 +384,16 @@ public void onError(@NonNull CameraDevice cameraDevice, int errorCode) {
         backgroundHandler);
   }
 
-  private void createCaptureSession(int templateType, Surface... surfaces)
-      throws CameraAccessException {
+  @VisibleForTesting
+  void createCaptureSession(int templateType, Surface... surfaces) throws CameraAccessException {
     createCaptureSession(templateType, null, surfaces);
   }
 
   private void createCaptureSession(
       int templateType, Runnable onSuccessCallback, Surface... surfaces)
       throws CameraAccessException {
     // Close any existing capture session.
-    closeCaptureSession();
+    captureSession = null;
```

Comment:
I think this is still necessary. 

> Creating a new capture session with [CameraDevice#createCaptureSession](https://developer.android.com/reference/android/hardware/camera2/CameraDevice#createCaptureSession(android.hardware.camera2.params.SessionConfiguration)) will close any existing capture session automatically, and call the older session listener's [StateCallback#onClosed](https://developer.android.com/reference/android/hardware/camera2/CameraCaptureSession.StateCallback#onClosed(android.hardware.camera2.CameraCaptureSession)) callback.


This plugin is using https://developer.android.com/reference/android/hardware/camera2/CameraDevice#createCaptureRequest(int) instead.

Reply:
`CameraDevice#createCaptureSession` is getting called further down in the method, see [here](https://github.com/flutter/plugins/blob/0cc5bd350911063f82b8900aae27741ee0628136/packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java#L459). As you'll see, for API 27 and below, the [deprecated version](https://developer.android.com/reference/android/hardware/camera2/CameraDevice#createCaptureSession(java.util.List%3Candroid.view.Surface%3E,%20android.hardware.camera2.CameraCaptureSession.StateCallback,%20android.os.Handler)) of `CameraDevice#createCaptureSession` is actually used (see [here](https://github.com/flutter/plugins/blob/0cc5bd350911063f82b8900aae27741ee0628136/packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java#L494)), but it too takes care of closing the old capture session because both ultimately call [this method](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/hardware/camera2/impl/CameraDeviceImpl.java#704) to close the session.

Reply:
Got it. Thank you